packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,251 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Safe
+import { GnosisSafe as Safe } from "safe-contracts/GnosisSafe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { OwnerManager } from "safe-contracts/base/OwnerManager.sol";
+
+// Interfaces
+import { ILivenessModule2 } from "interfaces/safe/ILivenessModule2.sol";
+
+/// @title LivenessModule2
+/// @notice This module allows for challenge-based ownership transfer to a fallback owner
+///         when the Safe becomes unresponsive. The fallback owner can initiate a challenge,
+///         and if the Safe doesn't respond within the challenge period, ownership transfers
+///         to the fallback owner.
+/// @dev This is a singleton contract. To use it:
+///      1. The Safe must first enable this module using ModuleManager.enableModule()
+///      2. The Safe must then configure the module by calling enableModule() with parameters
+contract LivenessModule2 is ILivenessModule2 {
+    /// @notice Configuration for a Safe's liveness module
+    struct SafeConfig {
+        uint256 livenessChallengePeriod;
+        address fallbackOwner;
+        uint256 challengeStartTime;
+    }
+
+    /// @notice Mapping from Safe address to its configuration
+    mapping(address => SafeConfig) public safeConfigs;
+
+    /// @notice Semantic version.
+    /// @custom:semver 2.0.0
+    string public constant version = "2.0.0";
+
+    /// @notice Enables the module by the multisig to be challenged and sets the liveness_challenge_period and
+    /// fallback_owner
+    /// @dev MUST set the caller as a safe
+    /// @dev MUST take as parameters liveness_challenge_period and fallback_owner and store them as related to the safe
+    /// @dev MUST accept an arbitrary number of independent safe contracts to enable the module
+    /// @param _livenessChallengePeriod The period in seconds for a liveness challenge
+    /// @param _fallbackOwner The address that will become owner if challenge succeeds
+    function enableModule(uint256 _livenessChallengePeriod, address _fallbackOwner) external {
+        if (_livenessChallengePeriod == 0 || _fallbackOwner == address(0)) {
+            revert LivenessModule2_InvalidParameters();
+        }
+
+        // Set the caller as a safe and store its configuration
+        SafeConfig storage config = safeConfigs[msg.sender];
+
+        // Store the parameters related to this safe
+        config.livenessChallengePeriod = _livenessChallengePeriod;
+        config.fallbackOwner = _fallbackOwner;
+        config.challengeStartTime = 0;
+
+        emit ModuleEnabled(msg.sender, _livenessChallengePeriod, _fallbackOwner);
+    }
+
+    /// @notice Disables the module by an enabled safe
+    /// @dev MUST only be executable by an enabled safe
+    /// @dev MUST erase the existing liveness_challenge_period and fallback_owner data related to the calling safe
```

Comment:
Worth noting in the natspec that disabling the module also cancels any ongoing challenges

Reply:
Fixed on https://github.com/ethereum-optimism/optimism/pull/17272/commits/989ae745c482be07f1377bcaa220a51fc70b4378